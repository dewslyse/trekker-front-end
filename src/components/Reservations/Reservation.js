import React, { useState } from 'react';
import DatePicker from 'react-datepicker';
import { useSelector, useDispatch } from 'react-redux';
import Dropdown from 'react-bootstrap/Dropdown';
import DropdownButton from 'react-bootstrap/DropdownButton';
import { useParams } from 'react-router-dom';
import { addNewReservation } from '../../store/actions/reservationActions';

import 'react-datepicker/dist/react-datepicker.css';

const Reservations = () => {
  const [startDate, setStartDate] = useState(new Date());
  const [title, setTitle] = useState();
  const destinations = useSelector((state) => state.destinations);

  const dispatch = useDispatch();

  const { id } = useParams();
  const createReservation = (e) => {
    e.preventDefault();
    const reservation = {
      destination_id: title,
      start_date: startDate.toLocaleDateString(),
      end_date: startDate.toLocaleDateString(),
      fee: 20.21,
    };
    dispatch(addNewReservation(JSON.stringify(reservation), reservation.destination_id));
  };

  return (
    <>
      <div
        className="r-c"
        style={{
          backgroundImage: `linear-gradient(
        325deg,
        rgba(87, 111, 1, 0.777) 0%,
        rgba(150, 191, 2, 0.93) 100%
        ),url(https://i.ibb.co/jftG089/safari.jpg)`,
          backgroundPosition: 'center',
          backgroundSize: 'cover',
        }}
      >
        <div className="r-d">
          <div className="r-h">
            <h1>Book your dream vacation</h1>
          </div>
          <div>
            <p>Select your dream location to enjoy the best holiday ever!</p>
            <form onSubmit={createReservation}>
              <div className="r-b">
                <DatePicker
                  className="dp-1"
                  selected={startDate}
                  value={startDate}
                  onChange={(newValue) => {
                    setStartDate(newValue);
                    console.log(startDate.toLocaleString())
                  }}
                />

                <DropdownButton align="end" title={title} id="down">
                  {destinations.map((destination) => {
                    if (destination.id == parseInt(id, 10)) {
                      return (<Dropdown.Item eventKey="1" onClick={() => setTitle(destination.id)}>{destination.city_name}</Dropdown.Item>);
                    }
                    return '';
                  })}
                </DropdownButton>

                <button type="submit" className="btn-1">Book now</button>
              </div>

            </form>

          </div>
        </div>
      </div>
    </>
  );
};

export default Reservations;
