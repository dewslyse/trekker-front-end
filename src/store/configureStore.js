import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import destinationReducer from './reducers/destinationReducer';
import reservationReducer from './reducers/reservationReducer';
import uiReducer from './reducers/uiReducers';
import userReducer from './reducers/userReducer';

const store = configureStore({
  reducer: {
    destinations: destinationReducer,
    reservations: reservationReducer,
    user: userReducer,
    ui: uiReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  devTools: process.env.NODE_ENV !== 'production',
});

export default store;
