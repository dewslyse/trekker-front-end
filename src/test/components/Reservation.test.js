import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../store/configureStore';
import Reservations from '../../pages/Reservation';

describe('Reservation Page', () => {
  it('renders correctly', () => {
    const tree = render(<Provider store={store}><Router><Reservations /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
