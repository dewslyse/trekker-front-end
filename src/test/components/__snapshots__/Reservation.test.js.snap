// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation Page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sidebar"
      >
        <img
          alt="logo"
          class="logo"
          src="trekkerlogo.png"
        />
        <ul
          class="sidebar_content"
        >
          <li
            class="page_link"
          >
            <a
              activeclassname="active-link"
              class="link"
              href="/Destinations"
            >
              DESTINATIONS
            </a>
          </li>
          <li
            class="page_link"
          >
            <a
              activeclassname="active-link"
              class="link"
              href="/Destinations"
            >
              RESERVE
            </a>
          </li>
          <li
            class="page_link"
          >
            <a
              activeclassname="active-link"
              class="link"
              href="/Destinations"
            >
              ADD SAFARI
            </a>
          </li>
          <li
            class="page_link"
          >
            <a
              activeclassname="active-link"
              class="link"
              href="/Destinations"
            >
              DELETE SAFARI
            </a>
          </li>
        </ul>
        <div
          class="links"
        >
          <button
            class="li"
            type="button"
          >
            <a
              class="enter"
              href="/login"
            >
              Login
            </a>
          </button>
          <br />
          <button
            class="li"
            type="button"
          >
            <a
              class="enter"
              href="/register"
            >
              Register
            </a>
          </button>
          <br />
        </div>
      </div>
      <div
        class="r-c"
        style="background-position: center; background-size: cover;"
      >
        <div
          class="r-d"
        >
          <div
            class="r-h"
          >
            <h3>
              Book your dream vacation
            </h3>
          </div>
          <div>
            <form
              method="post"
            >
              <div
                class="r-b"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="dp-1"
                      type="text"
                      value="08/25/2022"
                    />
                  </div>
                </div>
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="dp-1"
                      type="text"
                      value="08/25/2022"
                    />
                  </div>
                </div>
              </div>
              <button
                class="btn-1"
                type="submit"
              >
                Book now
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sidebar"
    >
      <img
        alt="logo"
        class="logo"
        src="trekkerlogo.png"
      />
      <ul
        class="sidebar_content"
      >
        <li
          class="page_link"
        >
          <a
            activeclassname="active-link"
            class="link"
            href="/Destinations"
          >
            DESTINATIONS
          </a>
        </li>
        <li
          class="page_link"
        >
          <a
            activeclassname="active-link"
            class="link"
            href="/Destinations"
          >
            RESERVE
          </a>
        </li>
        <li
          class="page_link"
        >
          <a
            activeclassname="active-link"
            class="link"
            href="/Destinations"
          >
            ADD SAFARI
          </a>
        </li>
        <li
          class="page_link"
        >
          <a
            activeclassname="active-link"
            class="link"
            href="/Destinations"
          >
            DELETE SAFARI
          </a>
        </li>
      </ul>
      <div
        class="links"
      >
        <button
          class="li"
          type="button"
        >
          <a
            class="enter"
            href="/login"
          >
            Login
          </a>
        </button>
        <br />
        <button
          class="li"
          type="button"
        >
          <a
            class="enter"
            href="/register"
          >
            Register
          </a>
        </button>
        <br />
      </div>
    </div>
    <div
      class="r-c"
      style="background-position: center; background-size: cover;"
    >
      <div
        class="r-d"
      >
        <div
          class="r-h"
        >
          <h3>
            Book your dream vacation
          </h3>
        </div>
        <div>
          <form
            method="post"
          >
            <div
              class="r-b"
            >
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <input
                    class="dp-1"
                    type="text"
                    value="08/25/2022"
                  />
                </div>
              </div>
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <input
                    class="dp-1"
                    type="text"
                    value="08/25/2022"
                  />
                </div>
              </div>
            </div>
            <button
              class="btn-1"
              type="submit"
            >
              Book now
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
